lib/mongo/server/monitor/connection.rb

```diff
@@ -131,7 +133,7 @@ def initialize(address, options = {})
         #
         # @since 2.0.0
         def timeout
-          @timeout ||= options[:connect_timeout] || CONNECT_TIMEOUT
+          @timeout ||= options[:connect_timeout]
```

Comment:
Hi @jarthod 

See the description [here](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#socket-timeout-for-monitoring-is-connecttimeoutms) in our Server Discovery and Monitoring specification.

This timeout is applied only to the monitor thread's connection. It will be used both when connecting and for when regularly executing the ismaster command. I've put the default value of CONNECT_TIMEOUT back in, as you're right that if there is no connect_timeout option set, it would be infinite when sending the command on the socket. (The default will have been used when connecting though in this line)

The server connection will use the connect_timeout from the options (or the default) when connecting but then the socket_timeout when sending the operation, if one is specified. If one is not specified, it will be infinite. 

Does that make sense?
EmilyHi @jarthod 

See the description [here](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#socket-timeout-for-monitoring-is-connecttimeoutms) in our Server Discovery and Monitoring specification.

This timeout is applied only to the monitor thread's connection. It will be used both when connecting and for when regularly executing the ismaster command. I've put the default value of CONNECT_TIMEOUT back in, as you're right that if there is no connect_timeout option set, it would be infinite when sending the command on the socket.

The server connection will use the connect_timeout from the options (or the default) when connecting but then the socket_timeout when sending the operation, if one is specified. If one is not specified, it will be infinite. 

Does that make sense?
Emily